src/ml_tooling/storage/artifactory.py

```diff
@@ -0,0 +1,114 @@
+from ml_tooling.storage import Storage, StorageEnvironment
+from ml_tooling.config import ConfigGetter
+from ml_tooling.utils import MLToolingError
+
+import os
+import pickle
+from io import BytesIO
+from typing import Tuple, Union, Optional, List
+from pathlib import Path
+from sklearn.base import BaseEstimator
+
+try:
+    from artifactory import ArtifactoryPath
+except ImportError:
+    raise MLToolingError("Artifactory not installed - run pip install dohq-artifactory")
+
+
+class ArtifactoryStorage(Storage):
+    """
+    Artifactory Storage class
+    """
+    def __init__(self,
+                 repo_path: Union[Path, str] = None,
+                 apikey: Optional[str] = None,
+                 auth: Optional[Tuple[str, str]] = None
+                 ):
+        self.repo_path = repo_path
+        self.auth = auth
+        self.apikey = apikey
+
```

Comment:
What if the user passes both?

Reply:
yes good point, i will also add some code that adds 'http://' to the start of the path if its not present

Reply:
supplying both is fine :)